sw/device/lib/dif/dif_otp_ctrl.h

```diff
@@ -0,0 +1,962 @@
+ * Interface.
+ *
+ * Whether this function or its 64-bit variant should be called is dependent on
+ * the most recent partition read from.
+ *
+ * @param otp An OTP handle.
+ * @param[out] value Out-param for the read value.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_result_t dif_otp_ctrl_dai_read32_end(const dif_otp_ctrl_t *otp,
+                                                  uint32_t *value);
+
+/**
+ * Gets the result of a completed 64-bit read operation on the Direct Access
+ * Interface.
+ *
+ * Whether this function or its 32-bit variant should be called is dependent on
+ * the most recent partition read from.
+ *
+ * @param otp An OTP handle.
+ * @param[out] value Out-param for the read value.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_result_t dif_otp_ctrl_dai_read64_end(const dif_otp_ctrl_t *otp,
+                                                  uint64_t *value);
+
+/**
+ * Schedules a 32-bit write on the Direct Access Interface.
+ *
+ * Writes are performed relative to a partition; `address` should be given
+ * relative to the start of `partition`. An error is returned for out-of-bounds
+ * access.
+ *
+ * Furthermore, `address` must be four-byte-aligned, and `partition` must not be
+ * a secret partition. An error is returned if neither condition is met.
+ *
+ * Note that this function cannot be used to program the digest at the end of a
+ * `SW` partition; `dif_otp_ctrl_dai_digest()` must be used instead.
+ *
+ * @param otp An OTP handle.
+ * @param partition The partition to program.
+ * @param address A partition-relative address to program.
+ * @param value The value to program into the OTP.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_dai_result_t dif_otp_ctrl_dai_program32(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition,
+    uint32_t address, uint32_t value);
+
+/**
+ * Schedules a 64-bit write on the Direct Access Interface.
+ *
+ * Writes are performed relative to a partition; `address` should be given
+ * relative to the start of `partition`. An error is returned for out-of-bounds
+ * access.
+ *
+ * Furthermore, `address` must be eight-byte-aligned, and `partition` must be
+ * a secret partition. An error is returned if neither condition is met.
+ *
+ * @param otp An OTP handle.
+ * @param partition The partition to program.
+ * @param address A partition-relative address to program.
+ * @param value The value to program into the OTP.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_otp_ctrl_dai_result_t dif_otp_ctrl_dai_program64(
+    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition,
+    uint32_t address, uint64_t value);
+
+/**
+ * Schedules a hardware digest operation on the Direct Access Interface.
+ *
+ * This operation will also lock writes for the given partition.
```

Comment:
It's worth putting **emph** on this in some fashion, it's a big gun.

Reply:
Done.